src/pdb.rs

```diff
@@ -512,13 +512,11 @@ impl DeviceSQLString {
         let (input, length) = nom::number::complete::le_u16(input)?;
         let (input, _) = nom::bytes::complete::tag(b"\x00")(input)?;
 
-        // TODO: Ensure that length is always even
-        //assert_eq!(length % 2, 0);
-        // Don't include the trailing two nullbytes in the string.
-        let str_length = usize::from(length) / 2 - 1;
+        // sanity check
+        debug_assert_eq!(length % 2, 0);
 
+        let str_length = usize::from(length - 4) / 2;
```

Comment:
Why do you subtract 4 here?

Reply:
Thats how its done in the reference. I don't know why these strings are off by 4 bytes, but they are. https://github.com/Deep-Symmetry/crate-digger/blob/d37097b68f186a1669c1e0358adf0cef254b9e52/src/main/kaitai/rekordbox_pdb.ksy#L945

Reply:
In that case, could you add a comment for this? Also, it's weird that we don't subtract 3 from the LongASCII string length then...

Reply:
That might be potential bug then. Looking at the large test db, the only LongASCII occurences are `LongASCII("\u{0}")`.

Reply:
Yep, pretty sure I found another error in the kaitai spec. The `LongASCII("\u{0}")` strings are actually not zero length, were just reading them one byte too early, so the string parsing stops at the null-byte. Looking at the kaitai spec, its taking that length and then reading one byte to early, and three bytes to far.